lib/supabase/files/updateFileName.ts

```
@@ -0,0 +1,76 @@
+import { isValidFileName } from "@/utils/isValidFileName";
+import { isValidStorageKey } from "@/utils/isValidStorageKey";
+import { isValidUUID } from "@/utils/isValidUUID";
+
+/**
+ * Checks if storage key contains control characters
+ * @param key Storage key to check
+ * @returns true if contains control characters, false otherwise
+ */
+function hasControlCharacters(key: string): boolean {
+  // Control characters are 0x00-0x1F and 0x7F-0x9F
+  return /[\x00-\x1F\x7F-\x9F]/.test(key);
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)